src/meta/src/hummock/manager/mod.rs

```diff
@@ -716,13 +719,17 @@ where
     }
 
     /// Cancels a compaction task no matter it's assigned or unassigned.
-    pub async fn cancel_compact_task(&self, compact_task: &mut CompactTask) -> Result<bool> {
-        compact_task.set_task_status(TaskStatus::Canceled);
+    pub async fn cancel_compact_task(
+        &self,
+        compact_task: &mut CompactTask,
+        task_status: TaskStatus,
```

Comment:
Maybe add an assertion to make sure `task_status` is one of those CANCELED statuses?

Reply:
Thanks, add assert in `cancel_compact_task_impl `